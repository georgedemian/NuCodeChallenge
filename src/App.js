import React, { useEffect, useState } from "react";
import SearchBar from "./components/searchbar";
import CardList from "./components/cardList";

const url = "https://api.github.com/repos/facebook/react/issues"

export default function App() {
  const [rawData, setRawData] = useState([]);
  const [query, setQuery] = useState(null);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const result = await fetch(url);
        const resultJson = await result.json();
        setRawData(resultJson);
      } catch (error) {
        throw new Error(error);
      }
    };
    fetchData();
  }, []);

  const filterList = () => {
    if (query) {
      return rawData.filter((data) =>
        data.title.toLowerCase().includes(query.toLowerCase())
      );
    } else {
      return [];
    }
  };

  return (
    <div className="App" style={{ width: "70%", margin: "10px auto" }}>
      <SearchBar id="searchbar" setQuery={setQuery} />
      <CardList list={filterList()} />
    </div>
  );
}
