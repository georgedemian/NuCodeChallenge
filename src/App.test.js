import React from 'react';
import { render, within } from '@testing-library/react';
import App from './App';
import SearchBar from './components/searchbar';
import { EnzymeAdapter } from 'enzyme';

describe('App', () =>{
  it('App match snapshot', ()=>{
    const app = <App />
    expect(app).toMatchSnapshot()
  })
})