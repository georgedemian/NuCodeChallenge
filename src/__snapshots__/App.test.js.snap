// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App has searchbar 1`] = `<App />`;

exports[`App App match snapshot 1`] = `<App />`;
